docs/spec_draft.md

```diff
@@ -3158,6 +3159,117 @@ More formally, `results[:][result_index] = reduce(input_slices)` where:
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.reduce_scatter
+
+### Semantics
+
+<img align="center" src="spec_draft/reduce_scatter.svg" />
+
+Within each process group in the StableHLO grid, performs reduction, using
+`computations`, over the values of the `operand` tensor from each process,
+splits the reduction result along `scatter_dimension` into parts, and scatters
```

Comment:
reword suggestion-
splits the ~~ reduction~~ result along `scatter_dimension` ~~into parts~~

Reply:
let us keep this open for the next reviewer. 

This wording is inspired by [AllToAll](https://github.com/openxla/stablehlo/blob/main/docs/spec_draft.md#semantics-4).

Reply:
@subhankarshah let me know if all your concerns are addressed. In that case this can be forwarded.